documentation/src/main/asciidoc/topics/con_performance_persistence.adoc

```diff
@@ -0,0 +1,38 @@
+[id='performance-persistence_{context}']
+= Persistence
+
+Configuring {brandname} to interact with a persistent data source has the most significant performance impact of any other capability.
+This performance penalty comes from the fact that more traditional data sources are inherently slower than in-memory caches.
+Read and write operations will always take longer when the call goes to non-volatile memory outside the JVM.
+
+Depending on how you use cache stores, though, the reduction of {brandname} performance is offset by the gains that come from being able to access data in memory.
+Persistent storage might slow down {brandname} but {brandname} still gives you a much faster data tier for your applications.
+Using persistent storage also gives other benefits such as allowing you to preserve state for graceful cluster shutdowns.
+You can also overflow data from your caches to persistent storage and gain capacity beyond what is available in memory only.
+For example, you can have ten million entries in total while keeping only two million of them in memory.
+
+{brandname} adds key/value pairs to caches and persistent storage in either write-through mode or write-behind mode.
+These writing modes have different impacts on performance so it is important to take them into account when planning a {brandname} deployment.
+
+[%autowidth,cols="1,1",stripes=even]
+|===
+|Writing mode | Effect on performance
+
+|**Write-through**
+|{brandname} writes data to the cache and persistent storage simultaneously, which increases consistency and avoids data loss that can result from node failure.
+
+The downside to write-through mode is that synchronous writes add latency and decreases throughput.
+`Cache.put()` calls result in application threads waiting until writes to persistent storage complete.
+
+|**Write-behind**
+|{brandname} synchronously writes data to the cache but then adds the modification to a queue so that the write to persistent storage happens asynchronously, which decreases consistency but reduces latency of write operations.
+
+When the number of write operations exceed the size of the modification queue, {brandname} uses the write-through mode until it can process all the writes in the queue.
+If you observe that throughput drops with write-behind, you can adjust the size of the modification queue to avoid reverting to synchronous writes.
+|===
```

Comment:
then you have passivation that ignores all of this table :rofl:

Reply:
probably worth mention that passivation affects read performance.

Reply:
I've added some details here. thanks @pruivo